src/relay/op/image/resize.cc

```diff
@@ -215,6 +225,8 @@ bool Resize3DRel(const Array<Type>& types, int num_inputs, const Attrs& attrs,
 
   const Resize3DAttrs* param = attrs.as<Resize3DAttrs>();
   ICHECK(param != nullptr);
+  ICHECK(param->size.size() == 3);
+  ICHECK(param->roi.size() == 10);
```

Comment:
Why 10 here? Similarly, where do 6 or 8 for 1D/2D come from? Doesn't onnx doc say this number is `rank x 2`? https://github.com/onnx/onnx/blob/master/docs/Operators.md#Resize You also use only 2/4/6 of rois in topi functions for 1D/2D/3D.

Reply:
Rank here includes N and C, for for 3D resize, rank is 5, and roi is len 10.

Reply:
I see. This means `roi` array is sensitive to the layout, and your change in `topi/image/resize.py` seems to support only NCW/NCHW/NCDHW. 

While the ONNX definition supports arbitrary rank and doesn't distinguish the positions of spatial axes, our op supports only certain layout + corresponding axes position. I think it is better to remove roi for batch and channel dims in the frontend (similar to how `out_size` is handled) and require `param->roi.size() == 6` here for better clarity and removing the need for dealing with roi layout in topi.

Reply:
Related to above, I wonder if it is safe hard code NCH/NCHW/NCDHW layout in the ONNX `Resize` op converter, when the spec doesn't care about layout... https://github.com/apache/tvm/blob/4c1b66f9379fed29988cefd7334f0cd4918c8e6a/python/tvm/relay/frontend/onnx.py#L2612-L2619. I remember the folklore of ONNX being tied to NCHW layout, but I don't find the official statement of that convention in the spec. 

Is this code supposed to work on models exported from TF?

Reply:
> I think it is better to remove roi for batch and channel dims in the frontend (similar to how out_size is handled) and require param->roi.size() == 6 here for better clarity and removing the need for dealing with roi layout in topi.

That makes sense, I'll work on that. 

> I remember the folklore of ONNX being tied to NCHW layout, but I don't find the official statement of that convention in the spec.

Convolution definitely has to be NCHW. 

https://github.com/onnx/onnx/blob/master/docs/Operators.md#inputs-2---3

The documentation isn't as expilict about it with Resize, but I haven't found a test or an example of anything else. 

I'm looking at the code. Nearest neighbor appears to go over all axes: https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/core/providers/cpu/tensor/upsample.cc#L136-L254

But bilinear is only supporting NCHW: https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/core/providers/cpu/tensor/upsample.cc#L1069-L1075